### PR TITLE
RELATED: RAIL-4066 Queries for catalog items

### PR DESCRIPTION
**Pull request by plavemar**
- improve catalog items availability
- prepare queries to fetch catalog items
- prepare selectors to check if catalog loaded

JIRA: RAIL-4066

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7443/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9294/ : FAILURE in 10m 03s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7446/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9298/ : SUCCESS in 16m 39s

File: libs/sdk-backend-tiger/src/backend/workspace/attributes/index.ts [hunk 1]
```diff
@@ -61,6 +71,26 @@ export class TigerWorkspaceAttributes implements IWorkspaceAttributesService {
...
+                const id = await objRefToIdentifier(ref, this.authCall);
+                return `attributes.id=${id}`;
+            })
+            .join(",");
+
+        return this.authCall(async (client) => {
+            const catalogItems = await loadAttributesAndDateDatasets(
+                client,
+                this.workspace,
+                filter.length < MAX_FILTER_LENGTH ? filter : "",
```
Comment (ivanmjartan) [inline libs/sdk-backend-tiger/src/backend/workspace/attributes/index.ts:86]: pls move this line into constant like:

const rsqlFilter = filter.length < MAX_FILTER_LENGTH ? filter : "";

and add coment to this line something like: when filers are too long then we fetch all of them and use this separate const in loadAttributesAndDateDatasets request.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7448/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9300/ : SUCCESS in 16m 29s

**Review by ivanmjartan — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2399

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2154/ : SUCCESS in 1m 03s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2440/ : SUCCESS in 19s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1262/ : SUCCESS in 13m 39s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9302/ : SUCCESS in 16m 05s

**Pull request merged**